### PR TITLE
Fix Openstack Node Source

### PR DESCRIPTION
**Pull request by nebilbenmabrouk**
- Openstack Node Source fixed
- New Openstack parameters are added
- Parameters descriptions and verifications fixed

### CODE REVIEW

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasController.java [hunk 1]
```diff
@@ -191,7 +216,7 @@ public String createAzureInfrastructure(String infrastructureId, String clientId
     }
 
     public Set<String> createInstancesWithPublicKeyNameAndInitScript(String infrastructureId, String instanceTag,
-            String image, int numberOfInstances, int hardwareType, String publicKeyName, List<String> scripts) {
+            String image, int numberOfInstances, String hardwareType, String publicKeyName, List<String> scripts) {
```
Comment (ShatalovYaroslav) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasController.java]: If now the parameter is  String hardwareType, then maybe it's not required any more:
String.valueOf(hardwareType) below?

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasJSONTransformer.java [hunk 2]
```diff
@@ -56,6 +56,32 @@ public static String getInfrastructureJSONWithEndPoint(String infrastructureId,
...
+
+        JSONObject infrastructure = new JSONObject().put("id", infrastructureId)
+                                                    .put("type", type)
+                                                    .put("credentials", credentials)
+                                                    .put("scope", scope)
+                                                    .put("region", region)
+                                                    .put("identityVersion", identityVersion)
+                                                    .put("toBeRemovedOnShutdown", toBeRemovedOnShutdown);
+
+        if (endpoint != null && !endpoint.isEmpty()) {
```
Comment (ShatalovYaroslav) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasJSONTransformer.java]: should it be the same check for all other passed String parameters?

File: infrastructures/infrastructure-openstack/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/OpenstackInfrastructure.java [hunk 3]
```diff
@@ -46,37 +46,52 @@
...
+
+    @Configurable(description = "Openstack identity version")
+    protected String identityVersion = null;
+
+    @Configurable(description = "Openstack image")
     protected String image = null;
 
     @Configurable(description = "Flavor type of OpenStack")
-    protected int flavor = 3;
+    protected String flavor = "3";
```
Comment (ShatalovYaroslav) [inline infrastructures/infrastructure-openstack/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/OpenstackInfrastructure.java]: should the value "3" be extracted as constant?

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasController.java (hunk 1 above)
Comment (nebilbenmabrouk) [reply infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasController.java]: You're right. Fixed, thanks

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasJSONTransformer.java (hunk 2 above)
Comment (nebilbenmabrouk) [reply infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasJSONTransformer.java]: Checking removed, thanks

File: infrastructures/infrastructure-openstack/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/OpenstackInfrastructure.java (hunk 3 above)
Comment (nebilbenmabrouk) [reply infrastructures/infrastructure-openstack/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/OpenstackInfrastructure.java]: "3" removed. Thanks

**Comment by nebilbenmabrouk** [comment]
re-test this please

**Pull request merged**